### PR TITLE
bpo-33783: Use proper class markup for random.Random docs

### PR DESCRIPTION
**Pull request by Carreau**
This should make it easier to spot the existence of this class

https://bugs.python.org/issue33783

### CODE REVIEW

File: Misc/NEWS.d/next/Documentation/2018-06-20-10-22-57.bpo-33783.9MeA9Q.rst
Comment (berkerpeksag) [inline Misc/NEWS.d/next/Documentation/2018-06-20-10-22-57.bpo-33783.9MeA9Q.rst]: We don't require a NEWS entry for small documentation patches.

File: Doc/library/random.rst
Comment (berkerpeksag) [inline Doc/library/random.rst]: Style nit: Please follow the existing style in `Doc/library/random.rst` and use 3-space indentation.

Comment (rhettinger) [reply Doc/library/random.rst]: Other than berker's nits, this looks fine and is ready to apply.  I added backport tags as well.

File: Misc/NEWS.d/next/Documentation/2018-06-20-10-22-57.bpo-33783.9MeA9Q.rst
Comment (Carreau) [reply Misc/NEWS.d/next/Documentation/2018-06-20-10-22-57.bpo-33783.9MeA9Q.rst]: I know, but then `bedevere/news`  mark the test as failing, in the all-PR view of GitHub the test appear as failing and then the PRs tends to not be reviewed... removed.

File: Doc/library/random.rst
Comment (Carreau) [reply Doc/library/random.rst]: Oops, yes, nice catch. done, and vimrc updated accordingly to not do that again.

**Comment by Carreau** [comment]
Comments addressed, ready for review.

File: Doc/library/random.rst
Comment (takluyver) [inline Doc/library/random.rst]: This wording suggests to me that the class is only useful as a base class for subclassing, whereas I think it's also useful to instantiate and use directly.

**Comment by Carreau** [comment]
I did not change the phrasing at all on purpose not to add another axis to
the discussion that can lead to disagreement. I'm happy to do so in this
PR, or in a separate one, with a slight preference to a separate one as
consensus a review seem to already been achieved on this.

**Comment by takluyver** [comment]
Oh, I'd missed that it was just moving a paragraph. Still, where the paragraph was before, there's a sentence just above it about instantiating `Random`, and then that paragraph says it can *also* be subclassed. The new arrangement separates those ideas, and if you only see the class description, it looks a bit like subclassing is the only option.

**Comment by Carreau** [comment]
I still see that as part as the module introduction, you can have multiple instance of this module by creating `Random` instances. I can re-add that to the class description but I dont' think it should be removed from the the top introduction.

**Comment by Carreau** [comment]
How's that ? Ive re-added the information in the class description.

**Review by takluyver — APPROVED** [review]

File: Doc/library/random.rst
Comment (rhettinger) [inline Doc/library/random.rst:314]: I would like to leave this information in the intro section.

Please just make a stub entry for ``.. class:: Random([seed])`` with just a note that this class implements the default psuedo-random number generator.

**Review by rhettinger — CHANGES_REQUESTED** [review]

File: Doc/library/random.rst
Comment (Carreau) [reply Doc/library/random.rst:314]: Like So ? That feel a bit short, but I'll trust you.

**Comment by Carreau** [comment]
I have made the requested changes; please review again

File: Doc/library/random.rst
Comment (rhettinger) [inline Doc/library/random.rst]: Replace "This class implement" with "Class that implements"

**Review by rhettinger — CHANGES_REQUESTED** [review]

**Comment by vstinner** [comment]
I removed the " needs backport to 3.6" label, the 3.6 branch no long accept bugfixes (only security fixes are accepted): https://devguide.python.org/#status-of-python-branches

**Comment by Carreau** [comment]
I have made the requested changes; please review again

**Review by vstinner — APPROVED** [review]
LGTM.

@rhettinger: Would you mind to review the update PR?

**Comment by Carreau** [comment]
Thanks @vstinner for the review. Looking forward for this being merge, I originally hopped it would not take that much time from core developers.

**Review by pganssle — APPROVED** [review]

**Comment by vstinner** [comment]
Hum, @miss-islington has some troubles to create the backport. Anyway, it might be worth it to check if the doc is generated as expected at https://docs.python.org/dev/library/random.html before merging a 3.7 backport :-)

**Comment by vstinner** [comment]
Ah, PR #12859 is the 3.7 backport.

**Pull request merged**
